drivers/flash/flash_npcx_fiu_nor.c

```diff
@@ -189,16 +195,32 @@ static int flash_npcx_nor_write_status_regs(const struct device *dev, uint8_t *s
 #if defined(CONFIG_FLASH_JESD216_API)
 static int flash_npcx_nor_read_jedec_id(const struct device *dev, uint8_t *id)
 {
+	int ret;
+
 	if (id == NULL) {
 		return -EINVAL;
 	}
 
-	return flash_npcx_uma_read(dev, SPI_NOR_CMD_RDID, id, SPI_NOR_MAX_ID_LEN);
+#if defined(CONFIG_ESPI_TAF)
```

Comment:
Consider use of runtime checks as opposed to precompiler for readability.
e..g  
if ( IS_ENABLED(CONFIG_ESPI_TAF) && espi_taf_npcx_block(espi_dev, true) != 0) {